sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -452,6 +452,15 @@ case class RegExpReplace(subject: Expression, regexp: Expression, rep: Expressio
   }
 }
 
+object RegExpExtract {
+  def checkGroupIndex(groupCount: Int, groupIndex: Int): Unit = {
+    if (groupCount < groupIndex) {
+      throw new AnalysisException(
```

Comment:
We do this check at runtime, so `AnalysisException` doesn't fit. How about `IllegalArgumentException`?

Reply:
OK. Thanks.

Reply:
Sorry, I missed that...